### PR TITLE
[v2] Code shuffle: raftServer in server.go, Raft in raft.go

### PR DESCRIPTION
**Pull request by ongardie-sfdc**
After this change:
 - raft.go consists of the public-facing API exported by the library (was mostly in api.go before).
 - server.go consists of the raftServer struct and its methods (was mostly in raft.go before),
   as well as the raftShared struct (was in state.go before).

Towards #172: "Merge state.go into raft.go?"

---

This will be annoying to review, sorry. I don't know of a better way. I didn't change any code, just moved things.

### CODE REVIEW

**Comment by ongardie-sfdc** [comment]
cc @superfell @slackpad

File: raft.go [hunk 1]
```diff
@@ -1,1470 +1,504 @@
...
-	startIndex Index
+	// ErrCantBootstrap is returned when attempt is made to bootstrap a
+	// cluster that already has state present.
+	ErrCantBootstrap = errors.New("bootstrap only works on new clusters")
+)
 
-	inflight      *list.List // list of logFuture in log index order
-	verifyBatches []verifyBatch
-}
+// This struct belongs to api.go in the implementation.
```
Comment (slackpad) [inline raft.go]: This comment is stale now wrt. "api.go".

**Comment by slackpad** [comment]
This looks good - there was one small comment change noted above which I noticed after merging.

**Comment by slackpad** [comment]
This was my cross-check :-)

image

**Comment by slackpad** [comment]
Tweaked that comment in https://github.com/hashicorp/raft/commit/89b248a0ff4f31eb9414532a75241c83057496f5.

**Comment by ongardie-sfdc** [comment]
Thanks @slackpad.

**Pull request merged**